### ISSUE TITLE
Support Deriving Via

### ISSUE THREAD

**Issue by ChrisPenner** [ennocramer/floskell#16]
Hey again! Not sure how comprehensive you want the tool to be; let me know if I'm reporting bugs for things you don't care about; but as it is, running the formatter breaks the module so it's probably worth mentioning :D 

Input:

```haskell
newtype MyType = MyType Int
  deriving (Semigroup, Monoid) via (M.Sum Int)
```

output:
```haskell
newtype MyType = MyType Int
    deriving via (M.Sum Int) (Semigroup, Monoid)
```

(Note that this doesn't compile)

Cheers!

**Comment by ennocramer** [comment]
Thanks for the report.  Basically, anything accepted by `haskell-src-exts` is supposed to work, though there are plenty dark corners with old/seldomly used parts of the language.  But is is definitely a bug. Also, fixed in `master`.

**Comment by ChrisPenner** [comment]
Cheers!

**Comment by sielicki** [comment]
I'm unable to build hie after this change


```
src/Floskell/Pretty.hs:1188:20: error:
    Not in scope: data constructor ‘DerivVia’
     |
1188 |             Just x@DerivVia{} -> (return (), space *> pretty x)
     |                    ^^^^^^^^
```

https://github.com/ennocramer/floskell/commit/13af3d9b1967a244c2661e31f0b8f8cb1e3a0f79#diff-222f18d77037a09571cdd6663bd8b847R1188

**Comment by ennocramer** [comment]
Apologies for the breakage. Fixed in `e88ac75`.

**Comment by sielicki** [comment]
Thank you for taking care of that so quickly. Cheers.
